### PR TITLE
Common Base for Link-Layer Update Model Operations

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This is a mechanical patch that draws some elements from #11683 into a base type in `networkingcommon`.

This new type will form the basis of a forthcoming model operation that updates link-layer data sourced from machine agents. In turn, this will allow removal of a great deal of convoluted logic from state, currently used for the same purpose.

The removal of unused interface methods and corresponding shrinkage in generated mock code makes this patch a net removal of code.

## QA steps

Changes are mechanical only. Unit tests for the `instancepoller` API facade remain green.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Link-Layer Common Model Operation Base](https://trello.com/c/8KkF2Rjd/2711-link-layer-common-model-operation-base)

File: apiserver/common/networkingcommon/linklayer.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	SetOriginOps(origin network.Origin) []txn.Op
+
+	// SetProviderNetIDsOps returns the transaction operations required to ensure
+	// that the input provider IDs are set against the address.
+	SetProviderNetIDsOps(networkID, subnetID network.Id) []txn.Op
+}
+
+// LinkLayerMachine describes a machine that can query
+// and update link-layer device data.
+type LinkLayerMachine interface {
```
Comment (achilleasa) [inline apiserver/common/networkingcommon/linklayer.go:54]: The name and docblock of this type seems confusing to me. Why not call it something like `MachineLinkLayerAccessor` or `MachineLinkLayerQuerier` etc? The method signatures indicate that whatever implements this interface only supports queries and not mutations.

File: apiserver/common/networkingcommon/linklayer.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	// layer-2 devices for the machine.
+	AllLinkLayerDevices() ([]LinkLayerDevice, error)
+
+	// AllAddresses returns all IP addresses assigned to the machine's
+	// link-layer devices
+	AllAddresses() ([]LinkLayerAddress, error)
+
+	// AssertAliveOp returns a transaction operation for asserting
+	// that the machine is currently alive.
+	AssertAliveOp() txn.Op
```
Comment (achilleasa) [inline apiserver/common/networkingcommon/linklayer.go:65]: This method seems to violate SRP as its purpose is different than the other methods in this interface. Any particular reason why this is not defined in a separate interface and then composed with this one in a super interface?

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
https://bugs.launchpad.net/juju/+bug/1885117
$$merge$$

**Comment by manadart** [comment]
`StatusSuite.TestStatusAllFormats` again. I've added this to https://bugs.launchpad.net/juju/+bug/1884918.
$$merge$$

**Pull request merged**